Matchmaking and similarity search frontend and HTTP API for backend.

levitation opened the pull request.

levitation committed:
Initial changes for matchmaking

levitation committed:
Change number_of_matchmaking_introductions_per_period into a dropdown so the user has no way to enter invalid values and get errors during form submission

levitation committed:
Similarity search frontend and HTTP API for backend. Similarity search algorithm connection is not implemented yet.

viktor-yunenko commented on eahub/base/static/components/search-profiles/search.html:
```diff
@@ -145,7 +145,7 @@ <h5 class="modal-title">Filters</h5>
                     <div class="row search__hit-summary" v-if="item.summary">
                       <h5>Summary</h5>
-                      <ais-snippet :hit="item" attribute="summary"/>
+                      <ais-snippet :hit="item" attribute="summary"/><!-- TODO roland: replace newlines with <br>? -->
```
FYI I think that isn't possible, algolia doesn't allow spaces in snippets, that's part of their indexing engine tech specs. And there can't be HTML tags in those fields. Could you remove those TODOs then?

levitation replied:
Can you please clarify, what do you mean "doesn't allow spaces in snippets"? Would newline characters, tab characters or non-breaking space characters be allowed? Maybe we could use them in place of HTML &lt;br&gt; tags? And then during rendering replace them with &lt;br&gt; tags?

viktor-yunenko commented:
Thank you Roland!

It generally makes sense. But could you remove your local env code to make the tests pass?

Also you don't need to add `  # roland` notes in code, we have inline git history that developers can enable in IDE when they want 👍

levitation committed:
Cleaning up unnecessary comments. Reverting settings.py and webpack.config.js to main project settings. Fix profiles\tests\test_profile_models.py

levitation committed:
Updating test_profile_models.py to account for new fields

levitation commented:
Thank you Victor for the quick review! 

Did the fixes you asked: 
- Reverted settings.py; 
- partially reverted webpack.config.js; 
- removed my name from comments on changed lines; 
- also adjusted test_profile_models.py a bit. 

Now the tests pass.
